src/components/implementation/no_interface/cFE_booter/osfilesys.h

```diff
@@ -1,113 +1,258 @@
+    while(path[strlen(path) - 1] == '/'){
+        path[strlen(path) - 1] = 0;
+    }
+
+    // iterate from right to left through the path until you find a '/'
+    // everything you have iterated through is the name of the file
+    for ( offset = path_len - 2 ; path[offset] != '/' && offset > 0 ; offset--) {
+        //do nothing
+    }
+    char *name = path + offset +1;
+
+    assert(0 < strlen(name));
+    return name;
+}
+
 /******************************************************************************
-** Tar Level Methods
+** fsobj Level Methods
 ******************************************************************************/
+int32 file_rm(struct fsobj *o);
 
-uint32 tar_load();
+int32 file_close_by_ino(int32 ino);
 
-uint32 tar_parse();
+int32 file_close_by_name(char *path);
 
-int32 tar_read(uint32 offset, char *buf, uint32 req_sz);
+uint32 newpart_get(struct f_part **part);
 
-uint32 tar_cphdr(uint32 tar_offset, struct fsobj *file);
+uint32 newfile_get(struct fsobj **o);
 
-uint32 round_to_blocksize(uint32 offset);
+uint32 file_insert(struct fsobj *o, char *path);
 
-static inline uint32 oct_to_dec(char *oct)
-{
-    int32 i, base;
-    int32 tot;
+int32 file_open(char *path, permission_t permission);
 
-    i = strlen(oct) - 1;
+int32 file_close(int32 filedes);
 
-    for (base = 1, tot = 0 ; i >= 0 ; i--, base *= 8) {
-        char val = oct[i];
+int32 file_read(int32 FD, void *buffer, uint32 nbytes);
 
-        assert(val <= '7' && val >= '0');
-        val = val - '0';
-        tot = tot + (val * base);
-    }
+int32 file_write(int32 FD, void *buffer, uint32 nbytes);
 
-    return tot;
-}
+struct fsobj *file_find(char *path);
 
-/******************************************************************************
-** fsobj Level Methods
-******************************************************************************/
+int32 file_create(char *path, permission_t permission);
 
-uint32 fsobj_init(struct fsobj *o);
+int32 file_remove(char *path);
 
-uint32 newfile_get(struct fsobj **o);
+int32 file_rename(char *old_filename, char *new_filename);
+
+int32 file_cp(char *src, char *dest);
+
+int32 file_mv(char *src, char *dest);
 
-uint32 file_insert(struct fsobj *o, struct FS *fs);
+int32 chk_fd(int32 FD);
 
-uint32 insert(struct fsobj *o, struct fsobj *root);
+int32 file_FDGetInfo(int32 FD, OS_FDTableEntry *fd_prop);
 
-int32 open(const char *path);
+int32 file_stat(char *path, struct hack_stat  *filestats);
 
-int32 close(int32 filedes);
+int32 file_lseek(int32 FD, int32 offset, uint32 whence);
 
-int32 read(int32  filedes, void *buffer, uint32 nbytes);
+permission_t permission_cFE_to_COS(uint32 permission);
 
-struct fsobj *find(const char *path, struct fsobj *root);
+uint32 permission_COS_to_cFE(permission_t permission);
+
+/******************************************************************************
+** dirent Level Methods
+******************************************************************************/
+int32 newfd_get(int32 ino);
```

Comment:
Getting rather confusing, have you considered just prefixing everything with `fs_`?